app/controllers/application_controller.rb

```diff
@@ -1,3 +1,14 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  before_filter :store_current_location, :unless => :devise_controller?
+
+  private
+  def store_current_location
+    store_location_for(:user, request.url)
+  end
```

Comment:
Yes, I know this is the old way, but I couldn't make it work the new way. 
So I added a chore in Piv to change it when time allows.